build.sbt

```diff
@@ -21,57 +21,127 @@ val Scala213 = "2.13.8"
 ThisBuild / crossScalaVersions := Seq(Scala213, "3.1.3")
 ThisBuild / scalaVersion := Scala213 // the default Scala
 
+val CatsVersion = "2.8.0"
+val CatsEffectVersion = "3.3.13"
+val MUnitVersion = "0.7.29"
+val MUnitCatsEffectVersion = "1.0.7"
+val OpenTelemetryVersion = "1.15.0"
+
+lazy val scalaReflectDependency = Def.settings(
+  libraryDependencies ++= {
+    if (tlIsScala3.value) Nil
+    else Seq("org.scala-lang" % "scala-reflect" % scalaVersion.value % Provided)
+  }
+)
+
 lazy val root = tlCrossRootProject
-  .aggregate(core, testkit, java)
+  .aggregate(
+    `core-common`,
+    `core-metrics`,
+    core,
+    `testkit-common`,
+    `testkit-metrics`,
+    testkit,
+    `java-common`,
+    `java-metrics`,
+    java
+  )
   .settings(name := "otel4s")
 
+lazy val `core-common` = crossProject(JVMPlatform, JSPlatform)
+  .crossType(CrossType.Pure)
+  .in(file("core/common"))
+  .settings(
+    name := "otel4s-core-common",
+    libraryDependencies ++= Seq(
+      "org.typelevel" %%% "cats-core" % CatsVersion,
+      "org.scalameta" %%% "munit" % MUnitVersion % Test
+    )
+  )
+
+lazy val `core-metrics` = crossProject(JVMPlatform, JSPlatform)
+  .crossType(CrossType.Pure)
+  .in(file("core/metrics"))
+  .dependsOn(`core-common`)
+  .settings(scalaReflectDependency)
+  .settings(
+    name := "otel4s-core-metrics",
+    libraryDependencies ++= Seq(
+      "org.typelevel" %%% "cats-effect" % CatsEffectVersion,
```

Comment:
Do we need full `IO` here, should it be ce-kernel or ce-std? Does it matter (it won't if we add fs2 deps). Do we care?

Reply:
I played with extending stream tracing to Natchez, and concluded that having a way to trace `Resource` was good enough for streams.  All a stream dependency brings is a convenience method around `Stream.resource(whateverCreatesTheSpan)`.

Reply:
Yeah, once a streaming scope is described as a `Resource`, the propagation works well with streams.

https://github.com/typelevel/otel4s/blob/d70a57d72ec57db00d26856703618922f041214f/java/src/test/scala/org/typelevel/otel4s/java/trace/TracerSuite.scala#L455